[sntp-client] simplify uint64 constants

abtink opened the pull request.

jwhui commented:
How about apply the same to SNTP-related code in [src/cli/cli.cpp](https://github.com/openthread/openthread/blob/49150943c194980751b72d43d791007d0fbe5a38/src/cli/cli.cpp#L2998)?

abtink commented:
@jwhui , I updated the commits. I also added a second commit to remove the `UINT32_MAX` definition from the `toolchain.h` (since with this change it will not be used in core anymore).
